### ISSUE TITLE
Parametes: [apipromotedGraphQLAPIIdOutput, authpromotedc0bdaafdUserPoolId, apipromotedGraphQLAPIEndpointOutput] do not exist in template

### ISSUE THREAD

**Issue by tomasyaya** [aws-amplify/amplify-cli#3530]
>**Note**: If your issue/bug is regarding the AWS Amplify Console service, please log it in the 
[Amplify Console GitHub Issue Tracker](https://github.com/aws-amplify/amplify-console/issues)

**Describe the bug**
When trying to push an update in the Api Amplify returns:

`UPDATE_FAILED      functionlinkTest AWS::CloudFormation::Stack Thu Feb 27 2020 11:46:02 GMT+0100 (Central European Standard Time) Parameters: [apipromotedGraphQLAPIIdOutput, authpromotedc0bdaafdUserPoolId, apipromotedGraphQLAPIEndpointOutput] do not exist in the template`

and

`Following resources failed

✖ An error occurred when pushing the resources to the cloud

Resource is not in the state stackUpdateComplete
There was an error pushing the API resource
ResourceNotReady: Resource is not in the state stackUpdateComplete`

**Amplify CLI Version**
V 4.13.3

**To Reproduce**
I just did a amplify pull, I deleted to types in the schema and the error prompt as described

**Expected behavior**
Let push the changes in schema.

**Desktop (please complete the following information):**
 - OS: [e.g. Mac]
 - Node Version. You can use `node -v` to check the node version on your system
Node v12.14.1

**Additional context**

Here is my "amplify/backend/function/linkTest/xxx-cloudrmation-template.json" parameters:

`"Parameters": {
		"env": {
			"Type": "String"
		},
		"authpromotedc0bdaafdUserPoolId": {
			"Type": "String",
			"Default": "authpromotedc0bdaafdUserPoolId"
		},
		"apipromotedGraphQLAPIIdOutput": {
			"Type": "String",
			"Default": "apipromotedGraphQLAPIIdOutput"
		},
		"apipromotedGraphQLAPIEndpointOutput": {
			"Type": "String",
			"Default": "apipromotedGraphQLAPIEndpointOutput"
		}
	},`

And the "amplify/backend/awscloudformation/nested-cloudformation-stack.yml" linkTest section:

`"functionlinkTest": {
			"Type": "AWS::CloudFormation::Stack",
			"Properties": {
				"TemplateURL": "https://s3.amazonaws.com/amplify-promoted-promodev-173925-deployment/amplify-cfn-templates/function/linkTest-cloudformation-template.json",
				"Parameters": {
					"authpromotedc0bdaafdUserPoolId": {
						"Fn::GetAtt": [
							"authpromotedc0bdaafd",
							"Outputs.UserPoolId"
						]
					},
					"apipromotedGraphQLAPIIdOutput": {
						"Fn::GetAtt": [
							"apipromoted",
							"Outputs.GraphQLAPIIdOutput"
						]
					},
					"apipromotedGraphQLAPIEndpointOutput": {
						"Fn::GetAtt": [
							"apipromoted",
							"Outputs.GraphQLAPIEndpointOutput"
						]
					},
					"env": "promodev"
				}
			}`

**Comment by SwaySway** [comment]
Hello @tomasyaya could you provide the steps you have taken that lead up to this point? 
We can try to replicate to further understand the root cause here.

**Comment by tomasyaya** [comment]
@SwaySway 

I first try to remove the linkTest function with function remove. It throw an error because it had dependencies with auth an api. I did and function update and remove the dependencies. When I tried to push I throw this error. Because of that, I did a pull env to "reset" everything. I change my api cause thats what I wanted and did a push and this error keeps prompting in.

The amplify meta json for the function goes as follow:

` "linkTest": {
            "service": "Lambda",
            "providerPlugin": "awscloudformation",
            "build": true,
            "dependsOn": [
                {
                    "category": "auth",
                    "resourceName": "promotedc0bdaafd",
                    "attributes": [
                        "UserPoolId"
                    ]
                },
                {
                    "category": "api",
                    "resourceName": "promoted",
                    "attributes": [
                        "GraphQLAPIIdOutput",
                        "GraphQLAPIEndpointOutput"
                    ]
                }
            ],`

I have tried some approaches described in similar issues such as:

1. Do an api update with default configuration
2. Delete backend/myproject/api/build and run amplify gql-compile
3. Remove everything but the AppSyncName from backend/api/myproject/parameters.json

**Comment by tomasyaya** [comment]
@SwaySway some update:

I updated the rest api and remove the path in with linkTest was being use.

I pushed and that went OK.

Afterwords I update the linkTest function and remove the dependencies to auth and api.

I pushed and same error: `UPDATE_FAILED      functionlinkTest AWS::CloudFormation::Stack Fri Feb 28 2020 10:08:28 GMT+0100 (Central European Standard Time) Parameters: [apipromotedGraphQLAPIIdOutput, authpromotedc0bdaafdUserPoolId, apipromotedGraphQLAPIEndpointOutput] do not exist in the template.`

I went to the amplify-meta.json and remove the depends on from the linkTest

Then I push and it went OK.

Then I tried to remove the linkTest function itself.

When I pushed an error occur:

`DELETE_FAILED amplify-promoted-promodev-173925-functionlinkTest-9CSFLKXLQZQ5 AWS::CloudFormation::Stack Fri Feb 28 2020 10:18:07 GMT+0100 (Central European Standard Time) The following resource(s) failed to delete: [LambdaExecutionRole].
DELETE_FAILED LambdaExecutionRole                                            AWS::IAM::Role             Fri Feb 28 2020 10:18:07 GMT+0100 (Central European Standard Time) Cannot delete entity, must detach all policies first. (Service: AmazonIdentityManagement; Status Code: 409; Error Code: DeleteConflict; Request ID: c7987026-09b4-43f8-952c-7901a436cec2)
⠼ Updating resources in the cloud. This may take a few minutes...

DELETE_FAILED   functionlinkTest AWS::CloudFormation::Stack Fri Feb 28 2020 10:18:10 GMT+0100 (Central European Standard Time) Embedded stack arn:aws:cloudformation:us-east-1:224389094907:stack/amplify-promoted-promodev-173925-functionlinkTest-9CSFLKXLQZQ5/78ebb050-5888-11ea-be1d-0a73682547f5 was not successfully deleted: The following resource(s) failed to delete: [LambdaExecutionRole].`

I quit the push (let me know if this can generate issues).

I went into the aws console to look for the linkTest lambda to remove the permissions of the role but to my surprise I the lambda is no more. Actually if I go to Amplify from the console, enter de project, go to backend and click on functions the loading spinner stays for ever.

So I did an amplify pull.

I try to update the API (modify the schema) and push, leaving the function issue for later.

But when I push problems again:

`Following resources failed

Resource Name: LambdaExecutionRole (AWS::IAM::Role)
Event Type: create
Reason: promotedLambdaRole3f544664-promodev already exists in stack arn:aws:cloudformation:us-east-1:224389094907:stack/amplify-promoted-promodev-173925-functionlinkTest-9CSFLKXLQZQ5/78ebb050-5888-11ea-be1d-0a73682547f5


Resource Name: p3tpxgqugjdtrnafmx2tit7wuuGraphQLSchema (AWS::AppSync::GraphQLSchema)
Event Type: update
Reason: Resource update cancelled
URL: https://console.aws.amazon.com/cloudformation/home?region=us-east-1#/stacks/arn%3Aaws%3Acloudformation%3Aus-east-1%3A224389094907%3Astack%2Famplify-promoted-promodev-173925-apipromoted-1WPROO23RZZ8V%2Fbf213660-51a5-11ea-8dd0-0ad6b76236b1/events


✖ An error occurred when pushing the resources to the cloud

Resource is not in the state stackUpdateComplete
There was an error pushing the API resource
ResourceNotReady: Resource is not in the state stackUpdateComplete`

Its trying to create the LambdaExcutionRole again. And I cannot seem to find the lambda itself from the console to try and fix it.

Let me know if this is clear or do you need more information

**Comment by tomasyaya** [comment]
@SwaySway Sorry for the short tale in this issue.

So I was able to track the LambdaExcutionRole in the stack of the cloud formation. I remove the permission and manually delete the Role.

Then I did a a function remove from the cli and deleted the linkTest function.

I pushed and everything OK.

Then I updated the API and push and . . . .  everything OK.

By now it's working.

Im seeing that some remove commands don't remove the pertinent information from the meta-json and that brings some issues.

Nevertheless, sorry for the inconvenience.
